### PR TITLE
Rewording and optimizing the Tests page in the Essential section of the documentation

### PR DESCRIPTION
**Pull request by shiroyasha**
Things that are not yet done:

- [x] Grammarly grammar checks
- [x] Making sure that links are leading to correct pages
- [x] Check if the sentences sound natural for native speakers
- [ ] Consider adding a CTA to add additional runners in the Test Results CLI
- [ ] Link to Test Results CLI README page 

---

### What are the changes and why am I introducing them?

There are two categories of changes:

- Changes that help with random access reading that are typical for reading docs.
- Changes related to the focus of the page. I'm eliminating parts that are related to "how it works internally", and introducing use cases and benefits instead.

As most of these changes are structural in nature, I'm providing an annotated image that aims to make it easier to review this PR.

Test Result Analysis

### CODE REVIEW

File: docs/essentials/test-summary.md [hunk 1]
```diff
@@ -1,110 +1,129 @@
 ---
-description: This guide shows you how to optimize your Semaphore 2.0 workflow for monorepo projects.
+description: This document
```
Comment (markoa) [inline docs/essentials/test-summary.md:2]: Unfinished. This generates the meta description for Google iirc

File: docs/essentials/test-summary.md [hunk 2]
```diff
@@ -1,110 +1,129 @@
...
-!!! beta "Feature in beta"
-    Beta features are subject to change.
+Semaphore collects XML test reports and uses them to provide insight into your
+pipelines.
 
-Working with tests in your projects should be efficient and easy. The bigger your project
-becomes, the harder it is to track all the failures in your workflows. Firstly
-you have to navigate to specific job and then search through the logs for failure messages:
+With test reports, you enable your team to get an effective and consistent view
+of your CI/CD test suite across different test frameworks. You get a clear
```
Comment (markoa) [inline docs/essentials/test-summary.md:11]: Perhaps: consistent view of your test suite across different test frameworks and stages in a CI/CD workflow?

File: docs/essentials/test-summary.md [hunk 3]
```diff
@@ -1,110 +1,129 @@
...
+Published test results are available on under the **Tests** tab.
 
-Job summary provides you with a dashboard where you can track your suites and quickly find failing specs.
+<img style="box-shadow: 0px 0px 5px #ccc" src="/essentials/img/test-summary/tests-tab.png" alt="Tests Tab on Workflow Page">
 
-![Test Summary Tab](img/test-summary/summary-tab.png)
+## Setting up test reports
 
-## How does it work?
+Semaphore collects JUnit XML reports from your jobs. Any test runner that can
```
Comment (markoa) [inline docs/essentials/test-summary.md:21]: Passive → Active voice

Semaphore supports any test runner...

File: docs/essentials/test-summary.md [hunk 4]
```diff
@@ -1,110 +1,129 @@
...
+      commands:
+         - [[ -f report.xml ]] && test-results publish report.xml
 ```
 
-This way if your test runner exits with non-zero status(due to the spec failure) report will be published.
+The `test-results` tool is part of [Semaphore's Toolbox]() and it is available
+in every Semaphore job.
+
+If you are running tests inside of a Docker container, see
+[how to use test-results CLI with Docker&nbsp;↗][working-with-docker]{target=_blank}
```
Comment (markoa) [inline docs/essentials/test-summary.md:69]: This is probably the first time I see use of the arrow. Also we generally do target blank.  If you want to introduce it let's make some somehow official rules and propagate to other pages? Personally I think a normal link would be fine.

**Review by markoa — CHANGES_REQUESTED** [review]
Love the big screenshot with explanations. Some details in a few sentences aside, I think that this is a great example of how to compose a docs page. I recommend we make it part of the playbook.

File: docs/essentials/test-summary.md (hunk 4 above)
Comment (shiroyasha) [reply docs/essentials/test-summary.md:69]: The ↗ was inertia on my part to make a change. It was already in the document when I started editing.
I don't any strong preference, but now that you pointed it out, I do generally feel that they are just a distraction.

Thanks for pointing it out. Removing it 🔪 from this doc and from several other places.

**Pull request merged**